### PR TITLE
Unify all Track and TrackList APIs

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
This PR was already reviewed and approved [here](https://github.com/videojs/video.js/pull/3744). This one will allow us to use browserstack to make sure that IE8-11 is not broken.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [x] Unit Tests updated or fixed
- [x] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by BrandonOCasey** [comment]
🎉  Yay, success!

**Comment by misteroneill** [comment]
💯

File: src/js/player.js [hunk 1]
```diff
@@ -822,22 +821,27 @@ class Player extends Component {
...
-    }, this.options_[techName.toLowerCase()]);
+    };
+
+    TRACK_TYPES.names.forEach((name) => {
+      const props = TRACK_TYPES[name];
+
+      techOptions[props.getterName] = this[props.privateName];
+    });
+
+    techOptions = assign(techOptions, this.options_[techName.toLowerCase()]);
```
Comment (gkatsev) [inline src/js/player.js]: assign mutates, so, we should be able to keep `techOptions` as `const`.

**Comment by BrandonOCasey** [comment]
@gkatsev made updates and removed all checks for `if (tracks)`. Had to update the defaultTech in the text-track tests to get everything to work.

**Pull request merged**